### DISCUSSION TITLE
GitHub Provider Email is null for private email addresses

### DISCUSSION THREAD

**Discussion by AlessandroVol23** [nextauthjs/next-auth#7932] [Help] [answered]
Hi 👋🏽

I am using the GitHub provider. 

With users who share their email addresses publicly, it works great. But for users who don't share their email address, it doesn't. 

Even it seems to be implemented [here](https://github.com/nextauthjs/next-auth/blob/a79774f6e890b492ae30201f24b3f7024d0d7c9d/packages/next-auth/src/providers/github.ts#L80)

I thought about doing the same call in a `session` callback but I don't have access to the `account` in this callback.

I'm on the latest `next-auth` version 4.22.1.

Could anybody solve this already? 

I found a similar issue [here](https://github.com/nextauthjs/next-auth/issues/374).

The docs also say that the email is always returned also when it is set to private
<img width="999" alt="image" src="https://github.com/nextauthjs/next-auth/assets/19362086/2c7df361-b8a3-42d1-a19d-6a5087670e43">

**Comment by AlessandroVol23** [comment, marked as answer]
Solved: The issue was the missing permissions in the github app for reading the user's email. I thought the permissions are only for the app and not the OAuth Scope 🤦🏽‍♂️

**Reply by Aycom366** [reply]
what's this missing permission if I may ask?

**Reply by Sim0n18** [reply]
You have to edit the settings in your application provider which you have create on github in the developer settings. Go into the edit mode of your application and find permissions and events and then Account Permissions. There you can select which permissions you need and what the user has to accept. Find E-Mail Adress and then you can select what you need. For the authorization, read-only is enough. Save the settings and restart the dev server.
